internal/proxy/impl.go

```diff
@@ -3945,3 +3944,111 @@ func (node *Proxy) GetImportState(ctx context.Context, req *milvuspb.GetImportSt
 	log.Info("received get import state response", zap.Int64("taskID", req.GetTask()), zap.Any("resp", resp), zap.Error(err))
 	return resp, err
 }
+
+// InvalidateCredentialCache invalidate the credential cache of specified username.
+func (node *Proxy) InvalidateCredentialCache(ctx context.Context, request *proxypb.InvalidateCredCacheRequest) (*commonpb.Status, error) {
+	ctx = logutil.WithModule(ctx, moduleName)
+	logutil.Logger(ctx).Debug("received request to invalidate credential cache",
+		zap.String("role", typeutil.ProxyRole),
+		zap.String("username", request.Username))
+
+	username := request.Username
+	if globalMetaCache != nil {
+		globalMetaCache.RemoveCredential(ctx, username) // no need to return error, though credential may be not cached
+	}
+	logutil.Logger(ctx).Debug("complete to invalidate credential cache",
+		zap.String("role", typeutil.ProxyRole),
+		zap.String("username", request.Username))
+
+	return &commonpb.Status{
+		ErrorCode: commonpb.ErrorCode_Success,
+		Reason:    "",
+	}, nil
+}
+
+// UpdateCredentialCache update the credential cache of specified username.
+func (node *Proxy) UpdateCredentialCache(ctx context.Context, request *proxypb.UpdateCredCacheRequest) (*commonpb.Status, error) {
+	ctx = logutil.WithModule(ctx, moduleName)
+	logutil.Logger(ctx).Debug("received request to update credential cache",
+		zap.String("role", typeutil.ProxyRole),
+		zap.String("username", request.Username))
+
+	credInfo := &credentialInfo{
+		username: request.Username,
+		password: request.Password,
+	}
+	if globalMetaCache != nil {
+		globalMetaCache.UpdateCredential(ctx, credInfo) // no need to return error, though credential may be not cached
+	}
+	logutil.Logger(ctx).Debug("complete to update credential cache",
+		zap.String("role", typeutil.ProxyRole),
+		zap.String("username", request.Username))
+
+	return &commonpb.Status{
+		ErrorCode: commonpb.ErrorCode_Success,
+		Reason:    "",
+	}, nil
+}
+
+func (node *Proxy) ClearCredUsersCache(ctx context.Context, request *internalpb.ClearCredUsersCacheRequest) (*commonpb.Status, error) {
+	ctx = logutil.WithModule(ctx, moduleName)
+	logutil.Logger(ctx).Debug("received request to clear credential usernames cache",
+		zap.String("role", typeutil.ProxyRole))
+
+	if globalMetaCache != nil {
+		globalMetaCache.ClearCredUsers(ctx) // no need to return error, though credential may be not cached
+	}
+	logutil.Logger(ctx).Debug("complete to clear credential usernames cache",
+		zap.String("role", typeutil.ProxyRole))
+
+	return &commonpb.Status{
+		ErrorCode: commonpb.ErrorCode_Success,
+		Reason:    "",
+	}, nil
+}
+
+func (node *Proxy) CreateCredential(ctx context.Context, req *milvuspb.CreateCredentialRequest) (*commonpb.Status, error) {
```

Comment:
Suggest adding a length limitation on username and password.

Reply:
yes, we have it

Reply:
password also need